tests/unit/mixins/run-test.js

```diff
@@ -124,6 +124,86 @@ test('runTask tasks can be canceled', function(assert) {
   }, 10);
 });
 
+test('scheduleTask invokes async tasks', function(assert) {
+  assert.expect(2);
+
+  let subject = this.subject();
+  let done = assert.async();
+  let hasRun = false;
+
+  run(() => {
+    subject.scheduleTask('actions', () => {
+      hasRun = true;
+      assert.ok(true, 'callback was called');
+      done();
+    });
+
+    assert.notOk(hasRun, 'callback should not have run yet');
```

Comment:
This assert makes sure that the scheduled task did not occur instantly after executing `subject.scheduleTask`.